test/test_distributions_mixture.py

```diff
@@ -78,6 +85,46 @@ def test_mixture_broadcast_batch_shape(
     _test_mixture(mixing_distribution, component_distribution)
 
 
+@pytest.mark.parametrize("batch_shape", [(), (1,), (7,), (2, 5)])
+@pytest.mark.parametrize("validate_args_for_normal", [False, True])
+@pytest.mark.parametrize("validate_args_for_half_normal", [False, True])
+@pytest.mark.filterwarnings(
+    "ignore:Out-of-support values provided to log prob method."
+    " The value argument should be within the support.:UserWarning"
+)
+def test_mixture_with_different_support(
+    batch_shape, validate_args_for_normal, validate_args_for_half_normal
+):
+    mixing_probabilities = jnp.ones(2) / 2
+    mixing_distribution = dist.Categorical(probs=mixing_probabilities)
+    component_distribution = [
+        get_normal(batch_shape, validate_args_for_normal),
+        get_half_normal(batch_shape, validate_args_for_half_normal),
+    ]
+    mixture = dist.MixtureGeneral(
+        mixing_distribution=mixing_distribution,
+        component_distributions=component_distribution,
+        support=True,
```

Comment:
`support` should be `constraints.real`. Maybe adding a check `is_instance(support, constraints.Constraint` in the mixture constructor to make sure that the argument has correct type.

Reply:
I have a question which is not relevant to this PR. I've noticed NumPyro doesn't use type annotations. Is this due to historical reasons, Python's dynamic typing, performance considerations, or building on other frameworks? I'm curious to learn more about the team's approach to code quality and maintainability.